noir-projects/aztec-nr/aztec/src/utils/bytes.nr

```diff
@@ -0,0 +1,120 @@
+// Converts the input bytes into an array of fields. A Field is ~254 bits meaning that each field can store 31 bytes.
+// This implies that M = ceil(N / 31).
```

Comment:
The `M` value is free however - is the caller expected to pass the correct one? It seems they could just e.g. use a lower `M` value and not convert some of the bytes.

Reply:
Added a check for [this](https://github.com/AztecProtocol/aztec-packages/pull/8590/commits/12b6dc9742d1f1f7029f1ebfc20be15bf043ab07). Can you confirm that it's a compile time check so should not add constraints?

I am asking because the test seems to correctly handle the comp time checks which is either a pleasant surprise (somewhat expected the test to just not compile) or it's not a comptime check.

Reply:
Is there no way to determine this automatically using numeric generic arithmetic?

Reply:
@jfecher is it in the realm of possibility to have rounding in arithmetic over generics?

Reply:
I think it will do integer division so you can use: 

```noir 
pub fn bytes_to_fields<let N: u32>(input: [u8; N]) -> [Field; (N+30) / 31] {
    let M: u32 = (N + 30) / 31;
    ...
```

Reply:
@Thunkar good idea! Will try implementing that in a PR up the stack to unblock this one and the NFT from merging. Thanks

Reply:
@benesjan yep, you can get rounding behavior with division or modulus.

We had a bug a little while ago where we were actually simplifying `N / M * M` to just `N` which is incorrect since we use integer division. E.g. `3 / 2 * 2` is `2`, not `3`. So also be aware if you do use division you can't cancel it out with a multiplication.